pkg/api/types/validation/record.go

```diff
@@ -164,20 +164,21 @@ func ValidateRecordValue(path *validation.Path, value *string, field *types.Fiel
 
```

Comment:
for this kind of switches, what do you think about having a mapping object/config somewhere like
`
validators: {
  FieldKindText: ValidateRecordStringValue,
  FieldKindReference: ValidateRecordReferenceValue,
  ...
}
`
which then would shorten this to `validators[fieldKind](path, value, field)`

Reply:
Absolutely. we should probably have an interface for field types. Because there are some stuff that fields always do
- Validate the record value
- Validate the form definition
- Convert to SQL representation
- Convert from SQL representation
- Add SQL fields/tables for storing the field
All that stuff would probably need to be under an interface such as
```
type FieldStrategy interface{
  ValidateRecordValue(value ...) validation.ErrorList
  ValidateFieldDefinition(fieldDefinition *types.FieldDefinition) validation.ErrorList
  ConvertToSQL(value ..) (interface{}, error)
  ConvertFromSQL(value ..) (FieldValue, error)
  PrepareSQLInfra(?) error
}

func GetStrategyForField(fieldDefinition *types.FieldDefinition) (FieldStrategy, error) {...}
```

Reply:
I'll create a ticket for it